From e0a56f339193e057415bfaf7cfea5106e5e97590 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=ADctor=20Manuel=20J=C3=A1quez=20Leal?=
 <vjaquez@igalia.com>
Date: Fri, 9 Jul 2021 11:01:14 +0200
Subject: [PATCH] Revert "vaapi: Demote vaapidecodebin to rank NONE."

This reverts commit 7a25c5d4ec95aefeca6515ac023b23c5dd330194.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer-vaapi/-/merge_requests/436>
---
 gst/vaapi/gstvaapi.c       |  2 +-
 gst/vaapi/gstvaapidecode.c | 12 ++++++------
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/gst/vaapi/gstvaapi.c b/gst/vaapi/gstvaapi.c
index 56ee703f..35326248 100644
--- a/gst/vaapi/gstvaapi.c
+++ b/gst/vaapi/gstvaapi.c
@@ -210,7 +210,7 @@ plugin_init (GstPlugin * plugin)
       GST_RANK_NONE, GST_TYPE_VAAPIPOSTPROC);
 
   gst_element_register (plugin, "vaapidecodebin",
-      GST_RANK_NONE, GST_TYPE_VAAPI_DECODE_BIN);
+      GST_RANK_PRIMARY + 2, GST_TYPE_VAAPI_DECODE_BIN);
 
   rank = GST_RANK_SECONDARY;
   if (g_getenv ("WAYLAND_DISPLAY"))
diff --git a/gst/vaapi/gstvaapidecode.c b/gst/vaapi/gstvaapidecode.c
index 4e8d35fb..76d1bc47 100644
--- a/gst/vaapi/gstvaapidecode.c
+++ b/gst/vaapi/gstvaapidecode.c
@@ -115,14 +115,14 @@ static const GstVaapiDecoderMap vaapi_decode_map[] = {
   {GST_VAAPI_CODEC_MPEG4, GST_RANK_PRIMARY, "mpeg4",
       "video/mpeg, mpegversion=4", NULL},
   {GST_VAAPI_CODEC_H263, GST_RANK_PRIMARY, "h263", "video/x-h263", NULL},
-  {GST_VAAPI_CODEC_H264, GST_RANK_PRIMARY + 1, "h264", "video/x-h264",
+  {GST_VAAPI_CODEC_H264, GST_RANK_PRIMARY, "h264", "video/x-h264",
       gst_vaapi_decode_h264_install_properties},
-  {GST_VAAPI_CODEC_VC1, GST_RANK_PRIMARY + 1, "vc1",
+  {GST_VAAPI_CODEC_VC1, GST_RANK_PRIMARY, "vc1",
       "video/x-wmv, wmvversion=3, format={WMV3,WVC1}", NULL},
-  {GST_VAAPI_CODEC_VP8, GST_RANK_PRIMARY + 1, "vp8", "video/x-vp8", NULL},
-  {GST_VAAPI_CODEC_VP9, GST_RANK_PRIMARY + 1, "vp9", "video/x-vp9", NULL},
-  {GST_VAAPI_CODEC_H265, GST_RANK_PRIMARY + 1, "h265", "video/x-h265", NULL},
-  {GST_VAAPI_CODEC_AV1, GST_RANK_PRIMARY + 1, "av1", "video/x-av1", NULL},
+  {GST_VAAPI_CODEC_VP8, GST_RANK_PRIMARY, "vp8", "video/x-vp8", NULL},
+  {GST_VAAPI_CODEC_VP9, GST_RANK_PRIMARY, "vp9", "video/x-vp9", NULL},
+  {GST_VAAPI_CODEC_H265, GST_RANK_PRIMARY, "h265", "video/x-h265", NULL},
+  {GST_VAAPI_CODEC_AV1, GST_RANK_PRIMARY, "av1", "video/x-av1", NULL},
   {0 /* the rest */ , GST_RANK_PRIMARY + 1, NULL,
       gst_vaapidecode_sink_caps_str, NULL},
 };
-- 
GitLab

